lib/client-connect.js

```diff
@@ -66,7 +62,10 @@ module.exports = function connect (client, opts, callback) {
   }
 
   try {
-    client[kEnqueue](new ConnectRequest(client, opts, callback))
+    client.dispatch({
+      ...opts,
```

Comment:
Can you please avoid `...`? It's performance has got slower in recent Node.js.

Reply:
Is object assign better? Also do you have a reference for the perf regression? Not important. Just curious.

Reply:
Fixed